openedx_external_enrollments/tests/external_enrollments/tests_pathstream_external_enrollments.py

```diff
@@ -0,0 +1,151 @@
+            )
+
+        get_enrollment_data_mock.assert_called_once_with(data)
+        model_mock.objects.get_or_create.assert_called_with(
+            controller_name='pathstream',
+            course_shell_id=self.course_id,
+            email=self.user_email,
+        )
+        external_enrollment_object.save.assert_called_once()
+        self.assertListEqual(external_enrollment_object.meta, meta_expected)
+
+    @patch('openedx_external_enrollments.external_enrollments.pathstream_external_enrollment.ExternalEnrollment')
+    @patch(
+        'openedx_external_enrollments.external_enrollments.'
+        'pathstream_external_enrollment.PathstreamExternalEnrollment._get_enrollment_data'
+    )
+    def test_post_enrollment_update_enrollment(self, get_enrollment_data_mock, model_mock):
+        """This test validates _post_enrollment method when an unenrollment event
+        is triggered for user and course, which have been used to create the initial
+        ExternalEnrollment object."""
+        initial_meta = [
+            {
+                'enrollment_data_formated': 'course-v1:test+CS102+2019_T3,t@email,2021-06-28 16:40:31.4569,true\n',
+                'is_uploaded': False,
+            },
+        ]
+        external_enrollment_object = Mock()
+        external_enrollment_object.meta = initial_meta
+        model_mock.objects.get_or_create.return_value = (external_enrollment_object, False)
+        data = {
+            'user_email': self.user_email,
+            'is_active': False,
+            'course_id': self.course_id,
+        }
+        unenrollment_data = 'course-v1:test+CS102+2019_T3,t@email,2021-06-29 16:50:31.456900,false\n'
+        get_enrollment_data_mock.return_value = unenrollment_data
+        meta_expected = initial_meta
+        meta_expected.append(
+            {
+                'enrollment_data_formated': unenrollment_data,
+                'is_uploaded': False,
+            },
+        )
+
+        self.base._post_enrollment(data)  # pylint: disable=protected-access
+
+        external_enrollment_object.save.assert_called_once()
+        self.assertListEqual(external_enrollment_object.meta, meta_expected)
+
+    def test_post_enrollment_with_missing_data(self):
+        """This test validates the _post_enrollment execution for missing data.
+        For instance, if email or course_id are not in data, this will raise a Key error."""
+        data = {
+            'is_active': False,
+            'user_email': self.user_email,
+        }
+        log1 = 'calling enrollment for [{}] with data: {}'.format(self.base.__str__(), data)
+        log2 = 'calling enrollment for [{}] with course settings: {}'.format(self.base.__str__(), None)
+        log3 = 'Failed to complete enrollment due to missing data: {}'.format(data)
+
+        with LogCapture(level=logging.INFO) as log_capture:
+            self.base._post_enrollment(data)  # pylint: disable=protected-access
+
+            log_capture.check(
```

Comment:
Is it really throwing all these logs in one call?  @Jacatove Or why do you test all logs with just one call?

Reply:
This test must only capture the error log. The other logs are tested in 'test_post_enrollment_new_enrollment', so I will remove them in this test, and the answer to your first question is yes. @Squirrel18